lib/string.fz

```diff
@@ -518,6 +518,50 @@ string ref : has_equality, hasHash string, ordered string, strings is
       (s * missing) + string.this
 
 
+  # Takes a predicate p that takes a codepoint and returns true or false. Splits
```

Comment:
I suggest dropping the first sentence of the comment as it does not add any valuable information for understanding the purpose of the feature. 

I think the special handling of multiple, neighboring codepoints should be possible to turn off. I.e. if we have a CSV with lots of empty fields.

Would it make sense for `p` -  instead of taking a single codepoint - to take a sequence of all of the following codepoints?